### PR TITLE
Update revault tx

### PR DESCRIPTION
**Pull request by darosior**
This updates `revault_tx` to the latest version and does some consequent cleanups.

Based on #144

### CODE REVIEW

**Comment by darosior** [comment]
Still marked as WIP cause upstream is not merged but should be in a reviewable state.

File: src/daemon/bitcoind/actions.rs [hunk 1]
```diff
@@ -323,6 +324,30 @@ fn mark_confirmed_spends(
...
 }
 
+fn mark_confirmed_cancels(
+    revaultd: &Arc<RwLock<RevaultD>>,
+    bitcoind: &BitcoinD,
+) -> Result<(), BitcoindError> {
+    let db_path = revaultd.read().unwrap().db_file();
+
+    for (db_vault, cancel_tx) in db_canceling_vaults(&db_path)? {
+        let cancel_tx = cancel_tx.into_psbt().extract_tx();
```
Comment (danielabrozzoni) [inline src/daemon/bitcoind/actions.rs:334]: I think `txid()` can be used here too. Can you include the fix in be83f4f1b6d90e92d62ad113dd2c2563b886f623?

Comment (darosior) [reply src/daemon/bitcoind/actions.rs:334]: Right, good catch

**Comment by darosior** [comment]
Rebased on top of #144

**Comment by darosior** [comment]
Rebased after upstream merge, this is now ready for review (of course two tests still fail because of #144 )

**Comment by danielabrozzoni** [comment]
utACK 8fbae96a74d751f309447f7b5e1a6360b5a46b91 I read the code multiple times, it seems fine, I didn't test it directly

**Pull request merged**